src/test/codegen/try_identity.rs

```diff
@@ -0,0 +1,17 @@
+// compile-flags: -C no-prepopulate-passes -Z mir-opt-level=2
```

Comment:
Does this also work without -Z flags? Is this directly exposed to users via `--release`? Or should that be done in a separate PR?

Reply:
No idea. =) `-Z mir-opt-level=2` needed to trigger inlining, which this test needs. cc @wesleywiser 

We could maybe special case inlining for some functions via attributes?

Reply:
I'm not sure the inline pass is stable enough to offer even via an opt-in attribute.

Reply:
@wesleywiser just to clarify, I meant `#[rustc_inline_me]` just for use in the standard library or something.

Reply:
@Centril Yeah, that's what I thought you meant. I just think the inliner has a lot of sharp edges and I'd rather not give people tools *this* sharp.

Reply:
@wesleywiser Fair enough; Is there a dumber "definitely correct" version of the inliner we could use for "just this function" which doesn't try to inline recursively and whatnot?

Reply:
Possibly yeah :)